packages/backend/src/managers/SnippetManager.ts

```diff
@@ -21,17 +21,38 @@ import { Request } from 'postman-collection';
 import { Publisher } from '../utils/Publisher';
 import { Messages } from '@shared/Messages';
 import type { RequestOptions } from '@shared/src/models/RequestOptions';
+import { quarkusLangchain4Jgenerator } from './snippets/quarkus-snippet';
+
+type Generator = (requestOptions: RequestOptions) => string;
 
 export class SnippetManager extends Publisher<Language[]> implements Disposable {
+  #languages: Language[] = getLanguageList();
+  #additionalGenerators = new Map<string, Generator>();
+
   constructor(webview: Webview) {
     super(webview, Messages.MSG_SUPPORTED_LANGUAGES_UPDATE, () => this.getLanguageList());
+    this.addVariant('java', 'Quarkus Langchain4J', quarkusLangchain4Jgenerator);
```

Comment:
could we have this in an init function ? We probably want to have some lazy loading, similar to the InferenceManager